### ISSUE TITLE
`manual_range_contains` not always raised

### ISSUE THREAD

**Issue by swy7ch** [rust-lang/rust-clippy#13775]
### Summary

Clippy does not raise warning in some cases when performing two tests to assess that a value is within a range. For now, it doesn't work with `iXX::abs()` or when the pattern looks like ` a <= 1 || a >=3` (which should propose `!(1..=3).contains(&a)`. I tested it with every integer type, i.e. with `i8`, `i16`, `i32`,` i64` and `isize`

Running `cargo clippy -- -Wclippy::pedantic` doesn't solve the issue.

### Lint Name

manual_range_contains

### Reproducer

I tried this code with `cargo clippy -- -Wclippy::pedantic`:

```rust
fn main() {
    let a: i16 = 2;
    let b: i16 = 4;
    let diff: i16 = b - a;
    if diff >= 1 && diff <= 3 {
        println!("The warning is raised.");
    }

    if diff.abs() >= 1 && diff.abs() <= 3 {
        println!("Here, it is not.");
    }

    let a: i64 = 2;
    let b: i64 = 4;
    let diff: i64 = b - a;
    if diff <= 1 || diff >= 3 {
        println!("The warning is not raised.");
    }

    if diff.abs() >= 1 && diff.abs() <= 3 {
        println!("Here, it is not either.");
    }
}
```

I expected to see this happen: raise a warning on every `if` expression because of `!manual_range_contains`

Instead, this happened: only the first `if` expression generates a warning

### Version

```text
rustc 1.83.0 (90b35a623 2024-11-26) 
binary: rustc 
commit-hash: 90b35a6239c3d8bdabc530a6a0816f7ff89a0aaf 
commit-date: 2024-11-26 
host: x86_64-unknown-linux-gnu 
release: 1.83.0 
LLVM version: 19.1.1
```

**Comment by swy7ch** [comment]
> If the pattern were instead a < 1 || a > 3, then the proposed alternative would be correct.

And it works as expected. I clearly didn't do my homework, and should have phrased things more carefully. Sorry for the inconvenience, and thank you for your patience.

I guess we can close this issue, I'm just not sure how: `Close as not planned`?

**Comment by swy7ch** [comment]
Well I'll simply close it, if a mod thinks otherwise they can always close it the proper way
